src/XRP/issued-currency-client.ts

```diff
@@ -813,20 +812,21 @@ export default class IssuedCurrencyClient {
    */
   public async redeemIssuedCurrency(
     sender: Wallet,
-    currency: string,
-    issuer: string,
-    amount: string,
+    issuedCurrency: IssuedCurrency,
   ): Promise<TransactionResult> {
     // Redemption of issued currency is achieved by sending issued currency directly to the original issuer.
     // However, the issuer field specified in the amount is treated as a special case in this circumstance, and should be
     // set to the address of the account initiating the redemption.
     // See: https://xrpl.org/payment.html#special-issuer-values-for-sendmax-and-amount
+    const specialIssuedCurrency: IssuedCurrency = {
+      currency: issuedCurrency.currency,
+      issuer: sender.getAddress(),
+      value: issuedCurrency.value,
+    }
```

Comment:
I think this might look cleaner like this. What do you think?

```javascript
const issuedCurrency: IssuedCurrency = {
  ...issuedCurrency,
  issuer: sender.getAddress(),
}
```